lib/_stream_readable.js

```diff
@@ -447,7 +447,8 @@ function maybeReadMore_(stream, state) {
 // for virtual (non-string, non-buffer) streams, "length" is somewhat
 // arbitrary, and perhaps not very meaningful.
 Readable.prototype._read = function(n) {
-  this.emit('error', new Error('not implemented'));
+  const err = new Error('not implemented');
+  process.nextTick(() => this.emit('error', err));
 };
```

Comment:
in v8 all functions are "hoisted" no matter where in a function they were declared. or at least that used to be the case in v8 unless they changed how that works